scenario_player/utils/reclaim.py

```diff
@@ -0,0 +1,520 @@
+        target_block=current_confirmed_head,
+    )
+
+    tracked_channels: Dict[ChannelID, Dict] = dict()
+
+    # Ignore closed channels and if an address is not under our control. since
+    # The withdraw will only work properly on open channel if performed by both
+    # participants.
+    for event in all_network_events:
+        is_useful_channel_open = (
+            event["event"] == ChannelEvent.OPENED
+            and event["participant1"] in address_to_candidate
+            and event["participant2"] in address_to_candidate
+        )
+        if is_useful_channel_open:
+            tracked_channels[event["channel_identifier"]] = event
+
+        is_useful_channel_close = event["event"] == ChannelEvent.CLOSED and (
+            event["channel_identifier"] in tracked_channels
+        )
+        if is_useful_channel_close:
+            del tracked_channels[event["channel_identifier"]]
+
+    log.debug("Channels found", channels=len(tracked_channels))
+
+    # There must be only one proxy per smart contract and and one JSONRPCClient
+    # per private key, this is necessary for transaction and nonce
+    # synchronization. This loops creates only the necessary instances. For
+    # this to work properly, this function cannot be executed concurrrently.
+    address_to_proxymanager: Dict[Address, ProxyManager] = dict()
+    for channel_open_event in tracked_channels.values():
+        for address in (channel_open_event["participant1"], channel_open_event["participant2"]):
+            if address not in address_to_proxymanager:
+                candidate = address_to_candidate[address]
+                client = JSONRPCClient(web3, candidate.privkey)
+                proxy_manager = ProxyManager(
+                    client,
+                    contract_manager,
+                    ProxyManagerMetadata(
+                        token_network_registry_deployed_at=None,
+                        filters_start_at=token_network_deployed_at,
+                    ),
+                )
+                address_to_proxymanager[address] = proxy_manager
+
+    pool = Pool()
+    greenlets = list()
+    for channel_open_event in tracked_channels.values():
+        for address in (channel_open_event["participant1"], channel_open_event["participant2"]):
+            candidate = address_to_candidate[address]
+            proxy_manager = address_to_proxymanager[address]
+            token_network = proxy_manager.token_network(
+                token_network_address, current_confirmed_head
+            )
+
+            greenlets.append(
+                pool.spawn(
+                    _withdraw_participant_left_capacity_from_channel,
+                    address_to_candidate=address_to_candidate,
+                    channel=channel_open_event,
+                    token_network=token_network,
+                )
+            )
+
+    # Wait until all transactions are mined, at this point we are ignoring
+    # errors.
+    pool.join()
```

Comment:
Changed to send all the transactions in parallel, this should have a big impact in wait time.